libs/crypto/jswrap_crypto.c

```diff
@@ -335,56 +338,120 @@ JsVar *jswrap_crypto_PBKDF2(JsVar *passphrase, JsVar *salt, JsVar *options) {
 static NO_INLINE JsVar *jswrap_crypto_AEScrypt(JsVar *message, JsVar *key, JsVar *options, bool encrypt) {
   int err;
 
-  unsigned char iv[16]; // initialisation vector
-  memset(iv, 0, 16);
+  size_t iv_size = 0;
+  unsigned char *iv_bytes;
 
   CryptoMode mode = CM_CBC;
+  unsigned int tagLength = 0;
+  
+  size_t additionalDataLen = 0;
+  
+  char *additionalDataPtr = NULL;
 
   if (jsvIsObject(options)) {
+    
     JsVar *ivVar = jsvObjectGetChild(options, "iv", 0);
-    if (ivVar) {
-      jsvIterateCallbackToBytes(ivVar, iv, sizeof(iv));
+    if (ivVar && !jsvIsUndefined(ivVar)) {
+      iv_size = (size_t)jsvGetLength(ivVar);
+      jsvIterateCallbackToBytes(ivVar, &iv_bytes, (unsigned int)iv_size);
```

Comment:
Sorry it's taken so long to look into this - but this one in particular looks broken.

iv_bytes is never set, so at this point jsvIterateCallbackToBytes will write into a random area of memory! It's possible that by using &iv_bytes you're just writing onto the stack, but there's definitely an issue here...